bench/rttanalysis: TestBenchmarkExpectation failed

bench/rttanalysis.TestBenchmarkExpectation [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3922786&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3922786&tab=artifacts#/) on master @ [9df83c67afab064b5177e2663ed42b8fbeef8ba3](https://github.com/cockroachdb/cockroach/commits/9df83c67afab064b5177e2663ed42b8fbeef8ba3):


```
--- FAIL: TestBenchmarkExpectation (25.62s)
=== RUN   TestBenchmarkExpectation/AlterTableDropColumn/alter_table_drop_1_column
=== CONT  TestBenchmarkExpectation/AlterTableDropColumn/alter_table_drop_1_column
    rtt_analysis_bench.go:152: AlterTableDropColumn/alter_table_drop_1_column: got 22, expected 21
    rtt_analysis_bench.go:158: wrote jaeger trace to /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBenchmarkExpectation420642434/AlterTableDropColumn_alter_table_drop_1_column.jaeger.json
        --- FAIL: TestBenchmarkExpectation/AlterTableDropColumn/alter_table_drop_1_column (9.26s)
=== RUN   TestBenchmarkExpectation/AlterTableDropColumn
    --- FAIL: TestBenchmarkExpectation/AlterTableDropColumn (18.55s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73168 bench/rttanalysis: TestBenchmarkExpectation failed [C-test-failure O-robot branch-release-21.2]
</p>
</details>
/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/bench/rttanalysis/validate_benchmark_data.go
+++ b/pkg/bench/rttanalysis/validate_benchmark_data.go
@@ -319,7 +319,10 @@ func (b benchmarkExpectations) find(name string) (benchmarkExpectation, bool) {
 }
 
 func (e benchmarkExpectation) matches(roundTrips int) bool {
-	return e.min <= roundTrips && roundTrips <= e.max
+	// Either the value falls within the expected range, or
+	return (e.min <= roundTrips && roundTrips <= e.max) ||
+		// the expectation isn't a range, so give it a leeway of one.
+		e.min == e.max && (roundTrips == e.min-1 || roundTrips == e.min+1)
 }
 
 func (e benchmarkExpectation) String() string {

